### ISSUE TITLE
Enable setting of lockdown for individual networks

### ISSUE THREAD

**Issue by brynmathias** [BDI-pathogens/OpenABM-Covid19#21]
Work networks lockdown scaling parameters are currently specified globally

To enable simulation of different types of lock downs, or enabling different groups to return to work / school, we need to set these parameters at a finer grained level

**Comment by shaunkhoo** [comment]
Seconding this request! I'm trying to implement this model for some epidemiological simulations and it would be useful to simulate a workplace-level sectoral lockdown.

**Comment by aaron-scheffler** [comment]
More dynamic control of lockdowns would be a great feature. In United States, we are transitioning from lockdown to a semi-lockdown. It would be wonderful if you could update the lockdown multipliers dynamically through the Python interface. For example.

t=0: infection commence
t=15: lockdown initiated (workplace, random multiplier = .2)
t = 30: lockdown partially lifted (update workplace, random multiplier = .5)

In this way, we can fluctuate the interaction multipliers to reflect open society, closed society, and partially open society. This will be critical in projecting need to testing, contact tracing, etc...

Many thanks!
Aaron

**Comment by srstevenson** [comment]
@aaron-scheffler If you want to update the lockdown parameters of the work, random, and household networks by a constant factor for all ages, you can already do this if you write your own loop over time steps. For example:
```python
import pandas as pd
from COVID19.model import Model, Parameters

params = Parameters(
    input_param_file="tests/data/baseline_parameters.csv",
    input_households="tests/data/baseline_household_demographics.csv",
)

model = Model(params)

results = []

for day in range(40):
    model.one_time_step()
    result = model.one_time_step_results()
    results.append(result)

    if day == 15:
        model.update_running_params("lockdown_on", 1)
    elif day == 30:
        model.update_running_params("lockdown_work_network_multiplier", 0.5)

df = pd.DataFrame(results)
```

**Comment by brynmathias** [comment]
Please see PR https://github.com/BDI-pathogens/OpenABM-Covid19/pull/24 where this is implemented for different age level networks.

Broader network control for sector based networks is under design

**Comment by aaron-scheffler** [comment]
Thank you for response. In the original code posted a few weeks ago, lockdown_work_network_multiplier was not in the set of parameters modifiable using model.update_running_params even if you add an exception it will not allow manually. 

However, with the current code as of May 7, 2020, you can modify occupation interactions using lockdown_occupation_multiplier_working_network (the name is no longer lockdown_work_network_multiplier). Because the documentation in covid19_model.pdf has not yet been updated to reflect these changes, it is unclear how the various lockdown_occupation parameters control interactions (e.g. on the primary and secondary networks which are also defined at the moment).

Regardless, I can now vary the reduction in interactions through different lockdown stages which is great and I look forward to sharing our implementation for California.

**Comment by brynmathias** [comment]
@aaron-scheffler we've just merged #24 you'll find that the `lockdown_work_network_multiplier` has now been split into 

```
lockdown_occupation_multiplier_primary_network	
lockdown_occupation_multiplier_secondary_network	
lockdown_occupation_multiplier_working_network
lockdown_occupation_multiplier_retired_network	
lockdown_occupation_multiplier_elderly_network
```

These are set to default flat values in the configuration file However you can update them with the following snippet:
`model.update_running_param("lockdown_occupation_multiplier_working_network", new_value)`

The names 
`lockdown_occupation_multiplier_primary_network` and `lockdown_occupation_multiplier_secondary_network` refer to the uk school systems `primary` and `secondary` level education

they refer to the age bins `0 -> 9` and `10 -> 19` respectivly

Hope that helps

**Comment by aaron-scheffler** [comment]
@brynmathias Yes this clarifies everything. Many thanks for your response and work on the codes.
